### ISSUE TITLE
'ID' might be exported but not stored in metadata

### ISSUE THREAD

**Issue by lukaszachy** [teemtee/tmt#1503]
I believe we have problem with generating 'id' as part of the test export (to polarion/nitrate). It is very easy to export & forget that git was modified so the only place where 'id' is stored is TCMS/polarion. (or plain git reset --hard to get rid of unwanted modifications)

If it is required then it should be created before AND committed to the git (same as other metadata changes).
If it is optional then do not create it as part of the export without explicit request.

Thoughts?

@psss @KwisatzHaderach @hegerj

**Comment by hegerj** [comment]
Personally I ignore `id` changes. I don't understand what it is for (even after asking), so I do not want it as a part of my tests.
Therefore I am strongly for the second option.

**Comment by KwisatzHaderach** [comment]
IMO the id should be required, but it's not entirely necessary, it's used mainly for exporting testruns to polarion for case searching (it's more precise than the other 2 methods: tcms case id or beaker-task)

This is literally the same as with extra-nitrate field, export generates it but it's not necessary right? well, no, another export would fail when extra-nitrate field is not there and yet there is no forcing the user to `git push`

**Comment by hegerj** [comment]
> This is literally the same as with extra-nitrate field, export generates it but it's not necessary right?

No. `extra-*` fields are generated during `import` and it is forced that they are pushed before `export`.

**Comment by KwisatzHaderach** [comment]
I've seen the code and done multiple imports, but am very interested where/how is this enforced.

**Comment by hegerj** [comment]
By the enforcing I meant `tmt.utils.validate_git_status()`, which doesn't check `extra-*` specifically, but it checks that the imported tests were pushed before export.

**Comment by KwisatzHaderach** [comment]
Exactly, so it works the same for id parameter.

**Comment by hegerj** [comment]
Yes but the difference is that all other parameters are generated during `import` so you are forced to push it if you want to `export`, but `id` is generated during `export` when I don't really care about it.

**Comment by hegerj** [comment]
I am still not convinced generating `id` by default is correct thing to do for our tests. There is no benefit and it causes confusion.

**Comment by psss** [comment]
> Can add the generation for TCMS import if it will be required, but considering TCMS is supposed to die at EOY, I don't think it's that important.

Yes. No significant time investments planned for nitrate integration in the near future. Closing. Feel free to reopen if you think this is still important enough.
